### DISCUSSION TITLE
Overriding values of non-primitive type in experiment

### DISCUSSION THREAD

**Discussion by DirkKuhn** [mit-ll-responsible-ai/hydra-zen#438] [Q&A] [answered]
Hi,

I am trying to override the activation function of my neural network in an experiment group. Unfortunately, I seem to always run into the error
```
hydra.errors.ConfigCompositionException: In 'experiment/selu': ReadonlyConfigError raised while composing config:
Cannot set value of read-only config node
    full_key: model.activation_fn.path
    object_type=Just
```
Here is the corresponding minimal example.
```python
from typing import Type
from torch import nn

from hydra_zen import to_yaml, store, builds, zen, launch, make_config
from omegaconf import DictConfig

class Model:
    def __init__(self, activation_fn: Type[nn.Module] = nn.ReLU):
        self.activation_fn = activation_fn

def app(zen_cfg: DictConfig, model: Model) -> None:
    print(to_yaml(zen_cfg, resolve=True))

store(Model, group='model')

Config = builds(
    app,
    populate_full_signature=True,
    hydra_defaults=[
        '_self_',
        {'model': 'Model'},
        {'experiment': 'selu'},
    ]
)

experiment_store = store(group='experiment', package='_global_')
experiment_store(
    make_config(
        hydra_defaults=['_self_'],
        model=dict(activation_fn=nn.SELU),
        bases=(Config,)
    ),
    name='selu'
)

if __name__ == '__main__':
    store.add_to_hydra_store()
    launch(Config, zen(app), version_base='1.3')
```
Is there anything I can do? I have already tried specifying the activation function in a separate config group. Is there maybe a way to make the node writable? Or do I have to pass the activation function as a partially instantiated object?
Thank you!

**Comment by rsokl** [comment]
Ah this maybe an unintentional side effect of `just`  returning a frozen dataclass: https://github.com/mit-ll-responsible-ai/hydra-zen/blob/796d8ac7c87c5ad245eb739cd27b81cf62eb19ca/src/hydra_zen/structured_configs/_implementations.py#L506

If that is the case, I will happily change it to unblock this workflow.

**Reply by rsokl** [reply]
I'll try to check this later today

**Reply by DirkKuhn** [reply]
Thanks, now it works!

**Reply by rsokl** [reply]
Cool, I should remove all frozen-by-default dataclasses to ensure that people don't hit this.

**Reply by rsokl** [reply]
@DirkKuhn  to clarify -- did you make this modification to the source code of hydra-zen locally on your end so that it works?

**Reply by DirkKuhn** [reply]
Yes, I only commented the line 506 out in my own virtual environment. Was I supposed to do something else?

**Reply by rsokl** [reply]
Nope! Just verifying before I move forward with making the fix in the library. Thanks!

**Reply by DirkKuhn** [reply]
Do you think you can make this modification the next couple of days? Or does it cause some issues?
Can you think of any disadvantages of passing the type instead of using partial configuration?
Again thanks so much!

**Reply by rsokl** [reply]
This should not cause any issues. I will try to get this in the latest pre-release tonight I hope.

**Reply by rsokl** [reply, marked as answer]
@DirkKuhn  `v0.11.0rc2` has been released and contains the fix to this issue. Thanks for your patience.

**Reply by DirkKuhn** [reply]
No worries. Thank you for your hard work :)
